src/sinks/databend/api.rs

```diff
@@ -0,0 +1,210 @@
+use std::collections::BTreeMap;
+
+use bytes::Bytes;
+use http::Request;
+use http::StatusCode;
+use hyper::Body;
+use serde::{Deserialize, Serialize};
+
+use crate::{http::Auth, http::HttpClient, sinks::util::UriSerde};
+
+use super::error::DatabendError;
+
+#[derive(Serialize, Debug)]
+pub(super) struct StageAttachment {
+    location: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    file_format_options: Option<BTreeMap<String, String>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    copy_options: Option<BTreeMap<String, String>>,
+}
+
+#[derive(Serialize, Debug)]
+pub(super) struct DatabendHttpRequest {
+    sql: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    stage_attachment: Option<StageAttachment>,
+}
+
+impl DatabendHttpRequest {
+    pub const fn new(sql: String) -> Self {
+        Self {
+            sql,
+            stage_attachment: None,
+        }
+    }
+
+    pub fn add_stage_attachment(
+        &mut self,
+        location: String,
+        file_format_options: Option<BTreeMap<String, String>>,
+        copy_options: Option<BTreeMap<String, String>>,
+    ) {
+        self.stage_attachment = Some(StageAttachment {
+            location,
+            file_format_options,
+            copy_options,
+        });
+    }
+}
+
+#[derive(Deserialize, Debug)]
+pub(super) struct DatabendHttpResponseSchemaField {
+    #[allow(dead_code)]
+    pub name: String,
+    #[allow(dead_code)]
+    pub r#type: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub(super) struct DatabendHttpResponseError {
+    pub code: u16,
+    pub message: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub(super) struct DatabendHttpResponse {
+    #[allow(dead_code)]
+    pub schema: Vec<DatabendHttpResponseSchemaField>,
+    pub data: Vec<Vec<String>>,
+    pub error: Option<DatabendHttpResponseError>,
+    pub next_uri: Option<String>,
+}
+
+#[derive(Debug)]
+pub(super) struct DatabendPresignedResponse {
+    #[allow(dead_code)]
+    pub method: String,
+    #[allow(dead_code)]
+    pub headers: BTreeMap<String, String>,
+    pub url: String,
+}
```

Comment:
Are these fields that are `allow(dead_code)` actually needed, it didn't look like they were getting used anywhere?

Reply:
They are used in integration test currently.

Reply:
We'd like to avoid the `dead_code` if possible, whether that's putting it behind a `cfg(test)` or just using those outside of the tests as well.

Reply:
changed to `#[cfg(all(test, feature = "databend-integration-tests"))]`